### PR TITLE
Added Javadocs in CombatSystem

### PR DESCRIPTION
**Pull request by billxu1401**

### CODE REVIEW

File: src/main/java/org/terasology/combatSystem/physics/systems/CombatForceSystem.java [hunk 1]
```diff
@@ -33,6 +33,9 @@ public void update(float delta) {
             Vector3f force = new Vector3f(body.force);
             force.div(body.mass);
             
+            /**
+             * Adds force to acceleration
```
Comment (Naman-sopho) [inline src/main/java/org/terasology/combatSystem/physics/systems/CombatForceSystem.java]: It is not a good practice to comment each line code this way especially if the code is self explanatory.
Moreover, use `//` for single comments.

File: src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+    /**
+     * 
+     * get the ultimate owner of the entity. 
+     * 
+     * @param entity  the entity of the ultimate owner
+     * @return  the ultimate owner
+     */
     public static EntityRef getUltimateOwner(EntityRef entity){
+        /**
+         * create a new instance of attacker
```
Comment (Naman-sopho) [inline src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java]: This is probably a bit "nitpicky" but you should try to start sentences with a capital letter, it looks better that way.

**Comment by billxu1401** [comment]
Capitalized some words and changed some unnecessary comments

File: src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java [hunk 3]
```diff
@@ -52,6 +65,11 @@ public static EntityRef getFirstOwner(EntityRef entity){
         }
     }
     
+    /**
+     * get all owners
```
Comment (pollend) [inline src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java]: that's already pretty apparent by the definition of the function. java docs should provide extra information that isn't apparent just by looking at the function call. 

example: https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/world/generation/World.java#L31

File: src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java [hunk 4]
```diff
@@ -38,6 +46,11 @@ private static EntityRef recursiveOwner(EntityRef entity){
         }
     }
     
+    /**
+     * Get first owner
```
Comment (pollend) [inline src/main/java/org/terasology/combatSystem/weaponFeatures/OwnerSpecific.java]: same thing as my last comment.

**Comment by billxu1401** [comment]
Classes documented:

GravitySystem
MeleeEvent
PrimaryAttackEvent
MeleeAttackSystem
OwnerSpecific

File: src/main/java/org/terasology/combatSystem/weaponFeatures/systems/MeleeAttackSystem.java [hunk 5]
```diff
@@ -10,20 +10,40 @@
...
 import org.terasology.math.geom.Vector3f;
 
+/**
+ * Manages meelee attacks
+ */
 @RegisterSystem
 public class MeleeAttackSystem extends BaseComponentSystem{
     
+    /**
+     * Launch a melee primary attack
```
Comment (jellysnake) [inline src/main/java/org/terasology/combatSystem/weaponFeatures/systems/MeleeAttackSystem.java]: This is already good enough but if you're making other changes to the PR, then this technically should be tweaked.

It does launch a melee attack, but more accurately it triggers when a PrimaryAttack is made with a melee weapon, and converts that to a melee attack

File: src/main/java/org/terasology/combatSystem/weaponFeatures/systems/MeleeAttackSystem.java [hunk 6]
```diff
@@ -10,20 +10,40 @@
...
         melee(entity, event.getTarget(), event.getWeaponLoc(), event.getTargetHitLoc());
     }
     
     //------------------------------private methods----------------------------
+    /**
+     * Damages the target if the target is in range
+     * @param entity The entity attacking
+     * @param target The target of the attack
+     * @param weaponLoc The location the attack is coming from in the world
+     * @param targetHitLoc The location the attack is hitting in the world
```
Comment (jellysnake) [inline src/main/java/org/terasology/combatSystem/weaponFeatures/systems/MeleeAttackSystem.java]: This is exactly the subtle difference between `targetLocation` and `targetHitLocation` I've pointed out in my other targets.

Here you got it right so I know you understand the difference ;)

**Comment by billxu1401** [comment]
@jellysnake  Documentations fixed :)

**Comment by casals** [comment]
Merging as per conversation history / changes checked - it seems that all outstanding requests were resolved.

**Pull request merged**